### PR TITLE
feat(codemod): add usechat input state transformation for v5

### PR DESCRIPTION
**Pull request by dancer**
## background

the v5 migration removes managed input state from the usechat hook. users need to manually manage input state with usestate instead of relying on the built-in input and handleinputchange from usechat. without this codemod, users would need to manually update every usechat usage in their codebase

## summary

- added codemod to transform usechat destructuring patterns
- removes input and handleinputchange from usechat
- adds usestate for local input management
- replaces handleinputchange with inline setters

## verification

tested with real-world patterns including aliased destructuring, partial extraction, and function-scoped replacements. all tests pass

## tasks

- [x] created replace-usechat-input-with-state codemod
- [x] handled import alias preservation
- [x] added function-scoped identifier replacement
- [x] created test fixtures for all edge cases
- [x] integrated into v5 upgrade bundle

### CODE REVIEW

File: packages/codemod/src/test/__testfixtures__/replace-usechat-input-with-state.input.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// @ts-nocheck
+import { useChat } from '@ai-sdk/react';
```
Comment (gr2m) [inline packages/codemod/src/test/__testfixtures__/replace-usechat-input-with-state.input.ts]: This codemod should also handle/upgrade imports from `ai/react`, to account for the possibility that the other codemod is not run first

https://ai-sdk.dev/docs/migration-guides/migration-guide-5-0#react-ui-hooks-moved-to-ai-sdkreact

Comment (gr2m) [reply packages/codemod/src/test/__testfixtures__/replace-usechat-input-with-state.input.ts]: addressed via 056cd8809 and d8389e1b1

**Review by gr2m — APPROVED** [review]
